### ISSUE TITLE
lsp: `types.d.ts` is interpreted as `types.d.ts.d.ts` and gives errors

### ISSUE THREAD

**Issue by Ciantic** [denoland/deno#16017]
I use simple `types.d.ts` file like this:

```
/// <reference types="./types.d.ts" />
```

But after a while the Deno Language Server tab in VSCode starts to give errors like:

```
Failed to request to tsserver Error: Could not find source file: 'file:///C:/Source/Deno/gutenberg-block-with-deno/src/types.d.ts.d.ts'.
```

Somehow it adds another `.d.ts` text at the end `types.d.ts.d.ts` and of course it does not exist.

Bug is only in LSP server because `deno` commands works fine, errors are also listed in "Deno Language Server tab" only.

Project [to test is here](https://github.com/Ciantic/gutenberg-block-with-deno).

deno --version
deno 1.26.0 (release, x86_64-pc-windows-msvc)
v8 10.7.193.3
typescript 4.8.3

**Comment by theowenyoung** [comment]
Same error on deno 1.31.2:

```
error: TS2300 [ERROR]: Duplicate identifier 'TheValue'.
  type TheValue = string | boolean | number;
       ~~~~~~~~
    at file:///Users/green/project/translate/userscript_type.d.ts.d.ts:43:8

    'TheValue' was also declared here.
      type TheValue = string | boolean | number;
           ~~~~~~~~
        at file:///Users/green/project/translate/userscript_type.d.ts:43:8
```

**Comment by bartlomieju** [comment]
@dsherret I believe this might have been fixed by https://github.com/denoland/deno/commit/8db853514caae431a0ce91360d854c1b7f3c405f ?

**Comment by dsherret** [comment]
I think the latest error in Deno 1.31.2 is unrelated to this issue. @theowenyoung see #18117, which will be fixed in 1.31.3

**Comment by nayeemrmn** [comment]
Our specifier normalization (which flattens things like `.d.ts.d.ts`) is more consistent now, and I can't reproduce. Closing for now.
